From 747c6207acf654665d23625537b962152922c8c5 Mon Sep 17 00:00:00 2001
From: Mehdi Sabwat <mehdisabwat@gmail.com>
Date: Mon, 24 Feb 2020 14:08:02 +0100
Subject: [PATCH 1/1] core: initial core build for emscripten, based on POSIX

posix/sort.c won't be added because qsort_r is not supported.
---
 src/Makefile.am | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/src/Makefile.am b/src/Makefile.am
index bf91159fcc..474ddcdd63 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -437,6 +437,17 @@ libvlccore_la_SOURCES += \
 	posix/timer.c
 endif
 
+if HAVE_EMSCRIPTEN
+libvlccore_la_SOURCES += \
+	posix/thread.c \
+	posix/getaddrinfo.c \
+	posix/error.c \
+	posix/dirs.c \
+	posix/filesystem.c \
+	posix/specific.c \
+	posix/timer.c
+endif
+
 if HAVE_DARWIN
 libvlccore_la_SOURCES += \
 	darwin/error.c \
@@ -481,6 +492,7 @@ if !HAVE_LINUX
 libvlccore_la_SOURCES += posix/wait.c
 endif
 if !HAVE_ANDROID
+if !HAVE_EMSCRIPTEN
 libvlccore_la_SOURCES += posix/sort.c
 if !HAVE_DARWIN
 libvlccore_la_SOURCES += \
@@ -503,6 +515,7 @@ endif
 endif
 endif
 endif
+endif
 
 if ENABLE_SOUT
 libvlccore_la_SOURCES += \
-- 
2.25.0

